conan/tools/cmake/cmakedeps/templates/target_data.py

```diff
@@ -322,7 +322,6 @@ def join_defines(values, prefix=""):
             # self.lib_paths = ""  IMPORTANT! LINKERS IN LINUX FOR SHARED MIGHT NEED IT EVEN IF
             #                      NOT REALLY LINKING LIB
             self.libs = ""
-            self.system_libs = ""
             self.frameworks = ""
```

Comment:
Are frameworks omitted as well? Because they shouldn't.

Reply:
The problem with frameworks is that there is no distinction between frameworks provided by the package and system frameworks. We cannot propagated and link package frameworks if the trait tell us we shouldn't, otherwise we would have incorrect linking. How many system frameworks are defined in conan-center recipes?

Reply:
There are only system frameworks. Many recipes have frameworks.

Reply:
In the meantime, it's quite obvious that cpp_info.frameworks are system frameworks in case of header-library.

Reply:
can we add traits for frameworks as well (e.g. `frameworks` and `transitive_frameworks`)?
CCI at the moment only uses system frameworks, but private projects actually package frameworks.

Reply:
Not necessary, the logic to propagate frameworks is the same as for libraries.

We are trapped in a bad situation, CC recipes have been using ``frameworks`` for system frameworks, while private projects have been using it also for package frameworks. There is no possible logic that will make this work without breaking something, so what we need to think now is a path forward, how to unlock this.

Reply:
Is there a `system_frameworks` attribute in cpp_info? I thought CCI recipes (and probably many private recipes) were using frameworks as "system_frameworks" because there was no alternative.

Reply:
I don't think we're trapped, we can add new attributes (e.g. `system_frameworks` and `custom_frameworks`) and slowly deprecate old ones (`frameworks`).
btw, frameworks are not exactly as libraries - they are like libraries + headers packaged together into directory. so their traits policy ideally should be a combination of `headers`/`transitive_headers` and `libraries`/`transitive_libraries`.

Reply:
We are trapped, because in the current propagation we cannot differentiate. We cannot always propagate frameworks, because that will do incorrect linking with package frameworks, and we cannot not always propagate frameworks, because system frameworks will not be there and fail to link. We would need to introduce new ``system_frameworks`` and migrate all ConanCenter recipes to make this work, but this will also be challenging and even breaking in 1.X, it will be difficult to backport this without breaking.

Reply:
Assuming system_frameworks is not backported to 1.X, recipes can still rely on `conan.conan_version` for these cases. A quick grep in conan-center shows 82 unique recipes with system frameworks in their package_info().